fedimint-server/src/consensus/server.rs

```diff
@@ -214,7 +211,76 @@ impl ConsensusServer {
         Ok((consensus_server, consensus_api))
     }
 
+    pub async fn run(&self, task_handle: TaskHandle) -> anyhow::Result<()> {
+        if self.cfg.consensus.broadcast_public_keys.len() == 1 {
+            self.run_single_guardian(task_handle).await
+        } else {
+            self.run_consensus(task_handle).await
+        }
+    }
+
+    pub async fn run_single_guardian(&self, task_handle: TaskHandle) -> anyhow::Result<()> {
+        assert_eq!(self.cfg.consensus.broadcast_public_keys.len(), 1);
+
+        while !task_handle.is_shutting_down() {
+            let session_index = self
+                .db
+                .begin_transaction()
+                .await
+                .find_by_prefix(&SignedBlockPrefix)
+                .await
+                .count()
+                .await as u64;
+
+            let mut item_index = self.build_block().await.items.len() as u64;
+
+            let session_start_time = std::time::Instant::now();
+
+            while let Ok(item) = self.submission_receiver.recv().await {
+                if self
+                    .process_consensus_item(
+                        session_index,
+                        item_index,
+                        item,
+                        self.cfg.local.identity,
+                    )
+                    .await
+                    .is_ok()
+                {
+                    item_index += 1;
+                }
+
+                // we rely on the module consensus items to notice the timeout
```

Comment:
Does it ever matter if we the network is quiet and this doesn't gets triggered for an extend period of time with items already in a current session?

Reply:
Possibly for recovery as one currently can't recover transactions that happened in an unfinished session. @joschisan

Reply:
Probably not a biggy, but why not wrap the `self.submission_receiver.recv()` in a timeout with a deadline to `sessions_end_time` and just do the right thing?